src/wav2vec2/tensorflow_addons.py

```diff
@@ -0,0 +1,324 @@
+import tensorflow as tf
+from typeguard import typechecked
+
+
+class Conv1DWithWeightNorm(tf.keras.layers.Conv1D):
+    """
+    tensorflow_addons.layers.WeightNormalization, torch.nn.WeightNorm gives different results
+    So it's better to implement it for our use case.
+    """
```

Comment:
It's great that you debugged this. Prefer filing an issue for this when you get a moment.

Reply:
This is also a great opportunity to add unit tests to verify that it behaves the way you expect.

Reply:
I have added the test to compare TF implementation with torch.